### ISSUE TITLE
Display dict of namedtuples

### ISSUE THREAD

**Issue by robertvazan** [google/python-fire#230]
Fire nicely prints a dict of dicts using one line per inner dict. When I try to do the same with dict of namedtuples (which otherwise behave like dicts after #161), I just get a help page listing all the dict keys. When I specify a dict key, then that namedtuple is printed. I would prefer all namedtuples in the dict to be printed when no key is specified.

**Comment by robertvazan** [comment]
BTW, it would be probably a good idea to display namedtuple itself like a dict, but then this somewhat depends on what namedtuples are commonly used for. Mine would be more readable with field labels.

**Comment by dbieber** [comment]
Good idea.

The relevant code to change is:

https://github.com/google/python-fire/blob/c830177ce00b1c6f123316e3cae81cbaec8377ca/fire/core.py#L258-L261

and https://github.com/google/python-fire/blob/c830177ce00b1c6f123316e3cae81cbaec8377ca/fire/value_types.py#L44

**Comment by armingh2000** [comment]
I've been working on this issue for a while.I think that namedtuples are not correctly handled yet. 
In inspectutils.py, the function IsNamedTuple is implemented wrong. Because namedtuple is not an instance of tuple ( isinstance(a namedtuple variable, tuple) is false). I wonder how it works on some namedtuples?
If we want to handle namedtuples, are we focusing on only collections module or also other modules as well?

**Comment by dbieber** [comment]
> Because namedtuple is not an instance of tuple ( isinstance(a namedtuple variable, tuple) is false).

I just confirmed in Python 3:

```
In [2]: import collections                                                                                                                                         
In [3]: Point = collections.namedtuple('Point', ['x', 'y'])                                                                                                        
In [4]: x = Point(1,2)                                                                                                                                             
In [5]: isinstance(x, tuple)
Out[5]: True
```

Where did you see a namedtuple that was not an instance of tuple?

> If we want to handle namedtuples, are we focusing on only collections module or also other modules as well?

Are there other collections or types you're thinking about? We don't have a clearly articulated scope. NamedTuples are important to handle well because they're commonly used.

**Comment by armingh2000** [comment]
> I just confirmed in Python 3:
> In [2]: import collections                                                                                                                                         
> In [3]: Point = collections.namedtuple('Point', ['x', 'y'])                                                                                                        
> In [4]: x = Point(1,2)                                                                                                                                             
> In [5]: isinstance(x, tuple)
> Out[5]: True

Yes. You're right. I made a mistake in passing the right variable to it.

> Are there other collections or types you're thinking about? We don't have a clearly articulated scope. NamedTuples are important to handle well because they're commonly used.

I've used typing.NamedTuple before. Which is a typed version of collection.namedtuple:(https://docs.python.org/3/library/typing.html#typing.NamedTuple)

It's Handled well here, like a collections.namedtuple object.

**Comment by armingh2000** [comment]
@dbieber 
I made a simple change and made a pull request. I think it'll do the job. Can you check?
